### PR TITLE
Make pairwise comparison chain more like LLM as a judge 

### PR DESCRIPTION
**Pull request by CG80499**

### CODE REVIEW

**Comment by hinthornw** [comment]
This one was the basis for the other string comparison chain here: https://github.com/langchain-ai/langchain/blob/master/libs/langchain/langchain/evaluation/comparison/prompt.py 

Would that suit your goals?

File: libs/langchain/langchain/evaluation/comparison/eval_chain.py [hunk 1]
```diff
@@ -236,15 +246,22 @@ def from_llm(
             ValueError: If the input variables are not as expected.
 
         """
+        if not (isinstance(llm, ChatOpenAI) and llm.model_name.startswith("gpt-4")):
```
[the remark this answers is not in the document]
Comment (CG80499) [reply libs/langchain/langchain/evaluation/comparison/eval_chain.py:249]: If they're using Azure we will now never give warning.

File: libs/langchain/langchain/evaluation/comparison/prompt.py [hunk 2]
```diff
@@ -5,64 +5,54 @@
...
-
-[RESPONSE B]
-{prediction_b}
-[/RESPONSE B]"""
+COMPARISON_TEMPLATE = ChatPromptTemplate.from_messages(
+    [
+        ("system", SYSTEM_MESSAGE),
+        (
+            "human",
+            "{criteria}{reference}[User Question]\n{input}\n\n\
```
[the remark this answers is not in the document]
Comment (CG80499) [reply libs/langchain/langchain/evaluation/comparison/prompt.py:35]: removed

File: libs/langchain/langchain/evaluation/comparison/eval_chain.py [hunk 3]
```diff
@@ -236,15 +247,24 @@ def from_llm(
             ValueError: If the input variables are not as expected.
 
         """
+        if not (
+            isinstance(llm, ChatOpenAI) and llm.model_name.startswith("gpt-4")
```
Comment (hinthornw) [inline libs/langchain/langchain/evaluation/comparison/eval_chain.py:251]: nit :
```suggestion
            isinstance(llm, (ChatOpenAI, AzureChatOpenAI)) and llm.model_name.startswith("gpt-4")
```

and remove the other check

**Review by hinthornw — APPROVED** [review]

**Pull request merged**